VisualizationBase/src/items/ViewItem.cpp

```diff
@@ -142,6 +130,63 @@ Model::Node* ViewItem::nodeAt(int column, int row) const
 	return nodes_[column][row];
 }
 
+void ViewItem::notifyAboutRemoval(Item *item)
+{
+	Q_ASSERT(item);
+	removeArrowsForItem(item);
+}
+
+void ViewItem::notifyAboutRemoval(Model::Node *node)
+{
+	Q_ASSERT(node);
+	//If somebody's spacing depends on the node or a child, remove
+	//the node's spacing target
+	QList<ViewItemNode*> spacingNodes;
+	for (auto node : allNodes())
```

Comment:
Ouch ouch, a local variable with the same name as the argument. This is asking for trouble. Please, rename this.